sbncode/CAFMaker/FillBlip.cxx

```diff
@@ -0,0 +1,104 @@
+#include "sbncode/CAFMaker/FillBlip.h"
+
+namespace caf
+{
+    void FillBlip(   const std::vector<blip::Blip>& LarBlips, std::vector<caf::SRBlip>& CAF_Blips)
+    {
+        int NBlips = LarBlips.size();
+        for(int iBlip=0; iBlip<NBlips; iBlip++)
+        {
+            blip::Blip CurrentBlip = LarBlips[iBlip];
+            caf::SRBlip NewBlip;
+            NewBlip.ID = CurrentBlip.ID;
+            NewBlip.isValid = CurrentBlip.isValid;
+            NewBlip.Cryostat = CurrentBlip.Cryostat;
+            NewBlip.TPC = CurrentBlip.TPC;
+            NewBlip.NPlanes = CurrentBlip.NPlanes;
+            NewBlip.MaxWireSpan = CurrentBlip.MaxWireSpan;
+            NewBlip.TimeTick = CurrentBlip.TimeTick;
+            NewBlip.Time = CurrentBlip.Time;
+            NewBlip.Charge = CurrentBlip.Charge;
+            NewBlip.Energy = CurrentBlip.Energy/1000.; //convert to GeV
+            NewBlip.EnergyESTAR = CurrentBlip.EnergyESTAR/1000.; //convert to GeV
+            NewBlip.EnergyPSTAR = CurrentBlip.EnergyPSTAR/1000.; //convert to GeV
+            NewBlip.ProxTrkDist = CurrentBlip.ProxTrkDist; 
+            NewBlip.ProxTrkID = CurrentBlip.ProxTrkID;
+            NewBlip.inCylinder = CurrentBlip.inCylinder;
+            NewBlip.Position = CurrentBlip.Position;
+            NewBlip.SigmaYZ = CurrentBlip.SigmaYZ;
+            NewBlip.dX = CurrentBlip.dX;
+            NewBlip.dYZ = CurrentBlip.dYZ;
+            if(CurrentBlip.truth.ID >= 0 ) //MC Blip
+            {
+                FillMCTruthBlip( CurrentBlip, NewBlip );
+            }
+            FillBlipRealtedHitCluster( CurrentBlip, NewBlip );
+            CAF_Blips.push_back(NewBlip);
+        }
+
+    }
+
+    void FillMCTruthBlip( blip::Blip& LarBlip, caf::SRBlip &CAF_Blip )
+    {
+      //CAF_Blip.truthBlip = LarBlip.truth;
+      CAF_Blip.truthBlip.ID = LarBlip.truth.ID;
+      CAF_Blip.truthBlip.Cryostat =LarBlip.truth.Cryostat;
+      CAF_Blip.truthBlip.TPC =LarBlip.truth.TPC;
+      CAF_Blip.truthBlip.Time =LarBlip.truth.Time;
+      CAF_Blip.truthBlip.TimeTick =LarBlip.truth.TimeTick;
+      CAF_Blip.truthBlip.DriftTime =LarBlip.truth.DriftTime;
+      CAF_Blip.truthBlip.Energy =LarBlip.truth.Energy;
+      CAF_Blip.truthBlip.DepElectrons =LarBlip.truth.DepElectrons;
+      CAF_Blip.truthBlip.NumElectrons =LarBlip.truth.NumElectrons;
+      CAF_Blip.truthBlip.LeadG4ID =LarBlip.truth.LeadG4ID;
+      CAF_Blip.truthBlip.LeadG4Index =LarBlip.truth.LeadG4Index;
+      CAF_Blip.truthBlip.LeadG4PDG =LarBlip.truth.LeadG4PDG;
+      CAF_Blip.truthBlip.LeadCharge =LarBlip.truth.LeadCharge;
+      CAF_Blip.truthBlip.Position =LarBlip.truth.Position;
+      CAF_Blip.truthBlip.Energy = CAF_Blip.truthBlip.Energy/1000.; //convert to GeV
+    }
+
+    void FillBlipRealtedHitCluster(blip::Blip& LarBlip, caf::SRBlip &CAF_Blip)
```

Comment:
Pass clusters to this function rather than the full blip objects, since clusters are the only object this function concerns with.